helix-core/src/main/java/org/apache/helix/task/TaskStateModel.java

```diff
@@ -283,6 +292,38 @@ public void reset() {
     }
   }
 
+  /**
+   * Loads Task and TaskFactory classes for command input
+   */
+  private boolean loadNewTask(String command) {
+    try {
+      // Read ZNRecord containing task definition information.
+      ZNRecord taskConfig = _manager.getHelixDataAccessor().getBaseDataAccessor()
+          .get("/" + _manager.getClusterName() + "/TASK_DEFINITION",
+              null, AccessOption.THROW_EXCEPTION_IFNOTEXIST);
+
+      // Open the JAR file and import Task(s) and TaskFactory classes.
+      File taskJar = new File(taskConfig.getSimpleField("JAR_FILE"));
+      URL taskJarUrl = taskJar.toURI().toURL();
+
+      // Import Task(s) and TaskFactory classes.
+      URLClassLoader classLoader = URLClassLoader.newInstance(new URL[]{taskJarUrl});
+      for(String taskClass : taskConfig.getListField("TASK_CLASSES")) {
+        classLoader.loadClass(taskClass);
+      }
+      Class cl = classLoader.loadClass(taskConfig.getSimpleField("TASKFACTORY"));
+
+      // Register the TaskFactory.
+      _taskFactoryRegistry.put(command, (TaskFactory) cl.newInstance());
+    } catch (MalformedURLException | ClassNotFoundException | InstantiationException
+        | IllegalAccessException | ZkNoNodeException | NullPointerException | HelixException e) {
```

Comment:
It's a code smell and not good style to catch so many types of exceptions. Also, catching a runtime exception like NullPointerException is awkward. A way to work around this is to break up your loadTask function into smaller methods and deal with different failure scenarios there.